src/main/java/build/buildfarm/server/services/ActionCacheService.java

```diff
@@ -96,24 +109,30 @@ public void onSuccess(@Nullable ActionResult actionResult) {
           @SuppressWarnings("NullableProblems")
           @Override
           public void onFailure(Throwable t) {
-            log.log(
-                Level.WARNING,
-                format(
-                    "getActionResult(%s): %s",
-                    request.getInstanceName(), DigestUtil.toString(actionDigest)),
-                t);
             Status status = Status.fromThrowable(t);
-            if (!call.isCancelled()) {
-              try {
-                responseObserver.onError(status.asException());
-              } catch (StatusRuntimeException e) {
-                // ignore
-              }
+            if (call.isCancelled()) {
+              cancellations.inc();
+              // no further logging/response required
+              return;
+            }
+            if (status.getCode() != Status.Code.CANCELLED) {
+              log.log(
+                  Level.WARNING,
+                  format(
+                      "getActionResult(%s): %s",
+                      request.getInstanceName(), DigestUtil.toString(actionDigest)),
+                  t);
+            }
+            try {
+              responseObserver.onError(status.asException());
```

Comment:
this was moved out of the "if" above. OK if responseObserver.onError(...) for Cancelled calls?